.pre-commit-config.yaml

```diff
@@ -19,4 +19,31 @@ repos:
         'flake8-bugbear==22.10.27',
         'flake8-comprehensions==3.10.1',
         'flake8-builtins==2.0.1'
+      ]
+
+-   repo: https://github.com/pre-commit/mirrors-mypy
+    rev: "v1.5.1"
+    hooks:
+    - id: mypy
+      entry: mypy ./weaviate
+      pass_filenames: false
+      additional_dependencies: [
```

Comment:
This is required because `pre-commit` runs in an isolated environment so `mypy` has no vision into all the dependencies referenced within our code without specifying them in `additional_dependencies` https://github.com/python/mypy/issues/13916

Without it, `mypy` complains about things like `requests.exceptions` stubs not existing

It'll be annoying to maintain this but then it is what DevOps is about... ;)

Reply:
> Without it, mypy complains about things like requests.exceptions stubs not existing

can't we just tell it to ignore that? Or would that hide errors?

Reply:
I think it would hide errors since part of using `mypy` is to test how your code's types align with the types of third-party libraries

Reply:
I was over-specifying here; only the stubs are required in this specification, have changed